crates/bevy_asset/src/io/file_asset_io.rs

```diff
@@ -199,13 +203,22 @@ pub fn filesystem_watcher_system(asset_server: Res<AssetServer>) {
                 for path in &paths {
                     let Some(set) = watcher.path_map.get(path) else {continue};
                     for to_reload in set {
-                        if !changed.contains(to_reload) {
-                            changed.insert(to_reload);
-                            let _ = asset_server.load_untracked(to_reload.as_path().into(), true);
-                        }
+                        // When an asset is modified, note down the timestamp (overriding any previous modification events)
+                        changed.insert(to_reload.to_owned(), Instant::now());
                     }
                 }
             }
         }
+
+        // Reload all assets whose last modification was at least 50ms ago.
+        //
+        // When changing and then saving a shader, several modification events are sent in short succession.
+        // Unless we wait until we are sure the shader is finished being modified (and that there will be no more events coming),
+        // we will sometimes get a crash when trying to reload a partially-modified shader.
+        for (to_reload, _) in changed.drain_filter(|_, last_modified| {
+            last_modified.elapsed() >= Duration::from_millis(50)
```

Comment:
this value will heavily depend on the setup of the user and should be configurable

Reply:
What's the best way to get the value from the plugin into the system?

Reply:
Store it in a resource, which is initialized by the plugin.

Reply:
No need for a resource here, you can add a parameter to the plugin under feature flag, pass it around when creating the `FileAssetIo` https://github.com/bevyengine/bevy/blob/main/crates/bevy_asset/src/lib.rs#L89, and store it there https://github.com/bevyengine/bevy/blob/main/crates/bevy_asset/src/io/file_asset_io.rs#L41. You already retrieve the `filesystem_watcher` in the watch system, this will be just next to it.

Reply:
@JMS55 implemented my suggestion in https://github.com/JMS55/bevy/pull/11